docs/reference/upgrade/close-ml.asciidoc

```diff
@@ -11,33 +11,36 @@ POST _ml/set_upgrade_mode?enabled=false
 // TEARDOWN
 ////////////
 
-If your {ml} indices were created earlier than the previous major version, they
-must be reindexed. In those circumstances, there must be no machine learning
-jobs running during the upgrade.
-
-In all other circumstances, there is no requirement to close your {ml} jobs.
-There are, however, advantages to doing so. If you choose to leave your jobs
-running during the upgrade, they are affected when you stop the {ml} nodes. The
-jobs move to another {ml} node and restore the model states. This scenario has
-the least disruption to the active {ml} jobs but incurs the highest load on the
-cluster.
-
-To close all {ml} jobs before you upgrade, see
-{stack-ov}/stopping-ml.html[Stopping {ml}]. This method persists the model
-state at the moment of closure, which means that when you open your jobs after
-the upgrade, they use the exact same model. This scenario takes the most time,
-however, especially if you have many jobs or jobs with large model states.
-
-To temporarily halt the tasks associated with your {ml} jobs and {dfeeds} and
-prevent new jobs from opening, use the <<ml-set-upgrade-mode,set upgrade mode API>>:
-
+If your {ml} indices were created before {prev-major-version}, you must stop
+all {dfeeds} and close all {ml} jobs then
+<<reindex-upgrade,reindex the indices>>.
```

Comment:
An alternative here is to temporarily halt the tasks associated with your {ml} jobs and {dfeeds} and prevent new jobs from opening by using the <<ml-set-upgrade-mode,set upgrade mode API>> during the reindex.

Reply:
Thanks, that step was added to the reindexing page in https://github.com/lcawl/elasticsearch/pull/1/files, so I've rebased and augmented the info there.

Reply:
You can preview the tip here: https://doc-review-63bf6.firebaseapp.com/es3/reindex-upgrade-inplace.html